### ISSUE TITLE
Presence penalty not implemented properly on ExLlamav2_HF

### ISSUE THREAD

**Issue by 11415142513152119** [oobabooga/text-generation-webui#4518]
### Describe the bug

Presence penalty on EXL2_HF does not change token probabilities regardless of value. Reported to work properly on cpp_HF so it's a problem with the EXL2_HF implementation.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

1. Load model using exl2_hf
2. Try to use presence penalty
3. Check logits

### Screenshot

![1](https://github.com/oobabooga/text-generation-webui/assets/146097564/a940fb92-0fb2-4e34-aee5-619a8ef7fdfe)
(At 2.00 value)

### Logs

```shell
not relevant
```

### System Info

```shell
not relevant
```

**Comment by 11415142513152119** [comment]
Never mind, apparently the effect is minimal on filled context even with 2 value.
